atom/src/validatebehavior.cpp

```diff
@@ -559,7 +559,8 @@ instance_handler( Member* member, CAtom* atom, PyObject* oldvalue, PyObject* new
         return 0;
     if( res == 1 )
         return newref( newvalue );
-    return py_type_fail( "invalid instance type" );
+    PyTypeObject* type = pytype_cast( member->validate_context );
```

Comment:
This is not necessarily correct since an `Instance` member can accept an iterable of types (contrary to a `Typed` member). Having a better error message makes sense but we need a dedicated function for it.

Reply:
Oops, thanks! I added one using repr, it's a bit verbose, but I think it should now handle all the difference cases isinstance accepts. I think it'd be acceptable to also just leave it out, just knowing which member of which object is usually enough to trace down the error.